hypertrace-view-generator/hypertrace-view-generator/src/test/java/org/hypertrace/viewgenerator/generators/SpanEventViewGeneratorTest.java

```diff
@@ -133,4 +139,64 @@ private void verifyCreateExitSpanToApiEntrySpan(
       EnrichedSpanUtils.isEntryApiBoundary(value);
     });
   }
+
+  private void verifyComputeApiExitCallCount_HotrodTrace(StructuredTrace trace) {
+    Map<ByteBuffer, Integer> eventToApiExitCallCount = spanEventViewGenerator.computeApiExitCallCount(trace);
+    ApiTraceGraph apiTraceGraph = new ApiTraceGraph(trace).build();
+    // this trace has 12 api nodes
+    // api edges
+    // 0 -> [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
+    // backend exit
+    // 1 -> to redis 13 exit calls
+    // 2 -> to mysql 1 exit call
+    // for events parts of api_node 0, there should 12 exit calls
+    // for events parts of api_node 1, there should be 13 exit calls
+    // for events parts of api_node 2, there should be 1 exit calls
+    Map<Integer, Integer> apiNodeToExitCallCount = Map.of(0, 12, 1, 13, 2, 1);
+    Map<ByteBuffer, Integer> eventToApiNodeIndex = buildEventIdToApiNode(apiTraceGraph);
+    eventToApiExitCallCount.forEach((k, v) -> {
+      Integer apiNodeIndex = eventToApiNodeIndex.get(k);
+      if (null != apiNodeIndex)
```

Comment:
Just curious: As part of this `verifyEveryEventPartOfSingleApiNode_HotrodTrace`, it verified that every event belongs to one of the ApiNode. So, can this be null?

Reply:
Ah! I can see from ApiTraceGraph code, there are some events as root with span kind - `UNKNOWN_SPAN_KIND_VALUE`. So, could be possible.